irods/password_obfuscation.py

```diff
@@ -275,3 +275,11 @@ def obfuscate_new_password(new, old, signature):
         new = new + padding[:lcopy]
 
     return scramble_v2(new, old, signature)
+
+
+def create_temp_password(temp_hash, source_password):
+    password = (temp_hash + source_password).ljust(100, chr(0))
```

Comment:
can you talk about this set of choices?    why 100?   why chr(0)?    is md5 sufficient in this case?

Reply:
This was based on the iRODS-php code. 
https://github.com/MaastrichtUniversity/irods-php/blame/master/src/RODSConn.class.php#L324

The earliest source I can find for that is here:
https://github.com/DICE-UNC/irods-php/blame/472dc46024e2122618a2d82daac03211396333a1/prods/src/RODSConn.class.php#L237

But I can also find it back here in Jargon:

https://github.com/DICE-UNC/jargon/blob/89b8b0bb86669b1e4dffc85b24e8f8abcc408b0c/jargon-core/src/main/java/org/irods/jargon/core/security/IRODSPasswordUtilities.java#L609

Isn't this simply how the iRODS server has implemented the passing of the temp password to a client? I presume to safely communicate the new secret password over a potential unencrypted connection. You pass just one part, and combine it with a secret the client has (the connection password).

The choice for MD5 is therefor also already made in the server. And yes it would be wise to change that to SHA. But I guess that's a protocol breaking change.

Reply:
very good - thanks.

the temp password behavior is a corner i've not visited before.

Reply:
Yes.. it's not well known. Maybe a bit of a shame, they are useful in this federated SAML access situation where the user does not have a (known) password. But they could also be used as short-lives access tokens as a sort of API token.